### ISSUE TITLE
Clang completer doesn't update completions on FileReadyToParse event

### ISSUE THREAD

**Issue by Tomeamis** [Valloric/ycmd#910]
As per the title. In the following file:

```
#include <iostream>

struct Hello
{
        bool b;
};

int main()
{
        Hello f;
        int foo;
        f.
}
```

If I send completions request just after the `f.`, then update the name of the bool to `boo` (sending this as a FileReadyToParse event), and ask for completions again, the server still sends `b` as a completion.
Oddly enough, if I ask for identifier completions in between the requests, the second clang request correctly returns `boo`.

Exact sequence of events:
Request completions with cursor just past `f.`
Send event updating `bool b;` to `bool boo;` and `f.` to `fo`
Request completions with cursor just past `fo` //1
Send event updating `fo` to `f.`
Request completions with cursor just past `f.` //2

If the sequence is executed as written here, the completion request //2 returns `boo` as completion. (As expected)
If the line //1 is left out, //2 returns `b` as completion.
This happens even when //1 is replaced with a wait, so this is not a timing issue.

Any insights would be greatly appreciated.

**Comment by bstaletic** [comment]
> Send event updating...

What do you mean by this? Our clang completer doesn't support renaming.

**Comment by micbou** [comment]
This is due to the caching mechanism. See issue https://github.com/Valloric/ycmd/issues/430.

**Comment by micbou** [comment]
This has been fixed in PR #958. Cache is now invalidated in that case.
